### PR TITLE
[3006.x] Add onedir artifacts to the download tests

### LINKED ISSUE #64246: [TECH DEBT] Onedir In Package Dowload Tests (Staging/Release)
### Description of the tech debt to be addressed, include links and screenshots
Add Onedir archives to the "Package downloads" tests for the staging and release pipelines.

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
Adds the onedir artifacts we publish to our suite of download tests

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/64246

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by MKLeb** [comment]
Passing staging run: https://github.com/saltstack/salt/actions/runs/5213112237/jobs/9407651751

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -220,8 +220,8 @@ jobs:
         run: |
           tools pkg repo publish release ${{ needs.prepare-workflow.outputs.salt-version }}
 
-  almalinux-8-pkg-download-tests:
-    name: Test Alma Linux 8 Package Downloads
+  almalinux-8-package-pkg-download-tests:
```
Comment (Ch3LL) [inline .github/workflows/release.yml:223]: Why are we naming these jobs `package-pkg` Seems we are duplicating information

**Review by twangboy — DISMISSED** [review]

**Review by Ch3LL — DISMISSED** [review]
just one small question, but shouldn't impact merge

File: .github/workflows/release.yml (hunk 1 above)
Comment (MKLeb) [reply .github/workflows/release.yml:223]: Ah, good catch, I can switch that to just be `package` or `onedir`

**Comment by MKLeb** [comment]
New staging run here: https://github.com/saltstack/salt/actions/runs/5215386251
No code changes, just workflow job renaming @Ch3LL.
Will remove WIP when it passes

**Review by Ch3LL — APPROVED** [review]

**Comment by MKLeb** [comment]
New passing staging run: https://github.com/saltstack/salt/actions/runs/5215386251

**Review by twangboy — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**